Services/Object/Icon/classes/class.ilObjectCustomIconFactory.php

```diff
@@ -20,17 +20,30 @@ class ilObjectCustomIconFactory
 	 * @var \ilObjectDataCache
 	 */
 	protected $objectCache;
+
+
+	/**
+	 * @var \ilDBInterface
+	 */
+	protected $dbInterface;
 	
 	/**
 	 * ilObjectCustomIconFactory constructor.
 	 * @param \ILIAS\Filesystem\Filesystem $webDirectory
 	 * @param \ILIAS\FileUpload\FileUpload $uploadService
 	 */
-	public function __construct(\ILIAS\Filesystem\Filesystem $webDirectory, \ILIAS\FileUpload\FileUpload $uploadService, \ilObjectDataCache $objectCache)
+	public function __construct(
+		\ILIAS\Filesystem\Filesystem $webDirectory,
+		\ILIAS\FileUpload\FileUpload $uploadService,
+		\ilObjectDataCache $objectCache,
+		\ilDBInterface $db
```

Comment:
Not required if a "lookup service" for resolved the target object is injected instead.